source/common/http/conn_manager_impl.cc

```diff
@@ -760,6 +760,12 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
                                connection_manager_.read_callbacks_->connection().dispatcher());
   request_headers_ = std::move(headers);
 
+  // TODO(alyssawilk) remove this synthetic path in a follow-up PR, including
+  // auditing of empty path headers.
+  if (HeaderUtility::isConnect(*request_headers_) && !request_headers_->Path()) {
```

Comment:
Isn't path guaranteed by the RFC to be empty? Should we just universally set it and ASSERT that it's empty?

Reply:
I believe it's allowed for HTTP/2
https://tools.ietf.org/html/draft-kinnear-httpbis-http2-transport-02

Reply:
Can you comment about the h2 thing?